code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -757,6 +759,8 @@ Pass a positive integer as an argument to override a bot's default speed.
 		qdel(src)
 
 /mob/living/simple_animal/bot/proc/topic_denied(mob/user) //Access check proc for bot topics! Remember to place in a bot's individual Topic if desired.
+	if(!Adjacent(user) && !issilicon(user))
+		return 1

```

Comment:
Good point. I'll just switch it to default canUseTopic not sure why i didn't use it actually.